### PR TITLE
feat: add "synchronized" event

### PR DESCRIPTION
**Pull request by ivarconr**
see #209 

Todo:

- [x] Add tests
- [x] Add promise wrapper for `initialize` method
- [x] Add `startUnleash()` documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -176,17 +171,18 @@ unleash.on('error', console.error);
...
-|            |
+| event        | payload                          | description                                                                                                                                                                                                                                  |
+| ------------ | -------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| ready        | -                                | is emitted once the fs-cache is ready. if no cache file exists it will still be emitted. The client is ready to use, but might not have synchronized with the Unleash API yet. This means the SDK still can operate on stale configurations. |
+| synchronized | -                                | is emitted when the SDK has successfully synchronized with the Unleash API and has alle the latest feature toggle configuration available.                                                                                                   |
```
Comment (SimenB) [inline README.md:177]:
```suggestion
| synchronized | -                                | is emitted when the SDK has successfully synchronized with the Unleash API and has all the latest feature toggle configuration available.                                                                                                   |
```

**Comment by ivarconr** [comment]
Hmz, seems like coveralls is not willing to send updated scores:
image

**Review by chriswk — APPROVED** [review]
Nice. Something to think about for our other SDKs as well?

**Comment by ivarconr** [comment]
> Nice. Something to think about for our other SDKs as well?

We have similar functionality in some of the other SDKs as well. E.g. in the Java SDK you can make it block until it has received the feature toggles from the API with the `synchronousFetchOnInitialisation` config option.

**Comment by FredrikOseberg** [comment]
> > Nice. Something to think about for our other SDKs as well?
> 
> We have similar functionality in some of the other SDKs as well. E.g. in the Java SDK you can make it block until it has received the feature toggles from the API with the `synchronousFetchOnInitialisation` config option.

There's also a WaitForClientReady() in the golang sdk, which I believe does the same thing.

**Review by FredrikOseberg — APPROVED** [review]
LGTM

**Comment by chriswk** [comment]
>We have similar functionality in some of the other SDKs as well. E.g. in the Java SDK you can make it block until it has received >the feature toggles from the API with the synchronousFetchOnInitialisation config option.

TIL

**Pull request merged**